### DISCUSSION TITLE
几个fcitx的问题

### DISCUSSION THREAD

**Discussion by Smart-SangGe** [fcitx/fcitx5#1151] [Q&A] [answered]
配置环境一会贴出 fcitx5-diagnose 的输出。现在遇到的问题有：1.输入法原本使用ctrl+space切换中英文，但是昨天不知道是更新了软件版本或者什么别的原因，现在仅能在alacritty中切换和输入中文。2.输入法常常在繁体和简体之间切换，印象中并没有按到什么快捷键。3.按照网上教程配置了emoji，但是好像并没有怎么出现，有时候会有unicode乱码出现在选单中。

fcitx5-diagnose
# System Info:
1.  `uname -a`:

        Linux redmi 6.11.2-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.11.2-1 (2024-10-05) x86_64 GNU/Linux

2.  `lsb_release -a`:

        Distributor ID:	Debian
        Description:	Debian GNU/Linux trixie/sid
        Release:	n/a
        Codename:	trixie

3.  `lsb_release -d`:

        Description:	Debian GNU/Linux trixie/sid

4.  `/etc/lsb-release`:

    `/etc/lsb-release` not found.

5.  `/etc/os-release`:

        PRETTY_NAME="Debian GNU/Linux trixie/sid"
        NAME="Debian GNU/Linux"
        VERSION_CODENAME=trixie
        ID=debian
        HOME_URL="https://www.debian.org/"
        SUPPORT_URL="https://www.debian.org/support"
        BUG_REPORT_URL="https://bugs.debian.org/"

6.  Desktop Environment:

    Desktop environment is `xfce`.

7.  XDG SESSION TYPE:

        XDG_SESSION_TYPE='x11'

8.  Bash Version:

        BASH_VERSION='5.2.32(1)-release'

# Environment:
1.  DISPLAY:

        DISPLAY=':0.0'

        WAYLAND_DISPLAY=''

2.  Keyboard Layout:

    1.  `setxkbmap`:

            xkb_keymap {
            	xkb_keycodes  { include "evdev+aliases(qwerty)"	};
            	xkb_types     { include "complete"	};
            	xkb_compat    { include "complete"	};
            	xkb_symbols   { include "pc+us+inet(evdev)"	};
            	xkb_geometry  { include "pc(pc105)"	};
            };

    2.  `xprop`:

            _XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us", "", ""

3.  Locale:

    1.  All locales:

            C
            C.utf8
            en_HK.utf8
            POSIX

    2.  Current locale:

            LANG=en_HK.UTF-8
            LANGUAGE=en_HK:en
            LC_CTYPE="en_HK.UTF-8"
            LC_NUMERIC="en_HK.UTF-8"
            LC_TIME="en_HK.UTF-8"
            LC_COLLATE="en_HK.UTF-8"
            LC_MONETARY="en_HK.UTF-8"
            LC_MESSAGES="en_HK.UTF-8"
            LC_PAPER="en_HK.UTF-8"
            LC_NAME="en_HK.UTF-8"
            LC_ADDRESS="en_HK.UTF-8"
            LC_TELEPHONE="en_HK.UTF-8"
            LC_MEASUREMENT="en_HK.UTF-8"
            LC_IDENTIFICATION="en_HK.UTF-8"
            LC_ALL=

4.  Directories:

    1.  Home:

            /home/sangge

    2.  `${XDG_CONFIG_HOME}`:

        Environment variable `XDG_CONFIG_HOME` is set to `/home/sangge/.config`.

        Current value of `XDG_CONFIG_HOME` is `~/.config` (`/home/sangge/.config`).

    3.  Fcitx5 Settings Directory:

        Current fcitx5 settings directory is `~/.config/fcitx5` (`/home/sangge/.config/fcitx5`).

5.  Current user:

    The script is run as sangge (1000).

# Fcitx State:
1.  executable:

    Found fcitx5 at `/usr/bin/fcitx5`.

2.  version:

    Fcitx version: `5.1.11`

3.  process:

    Found 1 fcitx5 process:

           5303 fcitx5

4.  `fcitx5-remote`:

    `fcitx5-remote` works properly.

5.  DBus interface:

    Using `dbus-send` to check dbus.

    Owner of DBus name `org.fcitx.Fcitx5` is `:1.93`.

    PID of DBus name `org.fcitx.Fcitx5` owner is `5303`.

    Debug information from dbus:

           Group [x11::0.0] has 3 InputContext(s)
          IC [9e05b08a2f59404bbbb045d4e2b3bd38] program:alacritty frontend:xim cap:4000000000 focus:1
          IC [5a861a825eb941d2b6f1472ccb08ffb5] program:alacritty frontend:xim cap:4000000000 focus:0
          IC [f259e6d5b530412c9ca5060b192c9e5a] program:alacritty frontend:xim cap:4000000000 focus:0
        Input Context without group

# Fcitx Configure UI:
1.  Config Tool Wrapper:

    Found fcitx5-configtool at `/usr/bin/fcitx5-configtool`.

2.  Config GUI for qt:

    Found `fcitx5-config-qt` at `/usr/bin/fcitx5-config-qt`.

3.  Config GUI for kde:

    **`kcmshell5` not found.**

# Frontends setup:
The environment variable checked by this script only shows the environment under current shell. It is still possible that you did not set the environment to the whole graphic desktop session. You may inspect the actual environment variable of a certain process by using `xargs -0 -L1 /proc/$PID/environ` for a certain process that you find not working.

## Xim:
1.  `${XMODIFIERS}`:

    Environment variable XMODIFIERS is set to "@im=fcitx" correctly.
    Xim Server Name from Environment variable is fcitx.

2.  XIM_SERVERS on root window:

    Xim server name is the same with that set in the environment variable.

## Qt:
1.  qt4 - `${QT4_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

    **`fcitx5-qt4-immodule-probing` not found.**

2.  qt5 - `${QT_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

    **`fcitx5-qt5-immodule-probing` not found.**

3.  qt6 - `${QT_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

    **`fcitx5-qt6-immodule-probing` not found.**

4.  Qt IM module files:

    Following error may not be accurate because guessing Qt version from path depends on how your distribution packages Qt. It is not a critical error if you do not use any Qt application with certain version of Qt or you are using text-input support by Qt under Wayland.
    **Cannot find fcitx5 input method module for Qt4.**
    **Cannot find fcitx5 input method module for Qt5.**
    **Cannot find fcitx5 input method module for Qt6.**

## Gtk:
1.  gtk - `${GTK_IM_MODULE}`:

    Environment variable GTK_IM_MODULE is set to "fcitx" correctly.

    **`fcitx5-gtk2-immodule-probing` not found.**

    **`fcitx5-gtk3-immodule-probing` not found.**

    **`fcitx5-gtk4-immodule-probing` not found.**

2.  `gtk-query-immodules`:

    1.  gtk 2:

        **Cannot find `gtk-query-immodules` for gtk 2**

        **Cannot find fcitx5 im module for gtk 2.**

    2.  gtk 3:

        **Cannot find `gtk-query-immodules` for gtk 3**

        **Cannot find fcitx5 im module for gtk 3.**

3.  Gtk IM module cache:

    1.  gtk 2:

        Found immodules cache for gtk `2.24.33` at `/usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules.cache`.
        Version Line:

            # Created by /usr/lib/x86_64-linux-gnu/libgtk2.0-0t64/gtk-query-immodules-2.0 from gtk+-2.24.33

        **Failed to find fcitx5 in immodule cache at `/usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules.cache`**

        **Cannot find fcitx5 im module for gtk 2 in cache.**

    2.  gtk 3:

        Found immodules cache for gtk `3.24.43` at `/usr/lib/i386-linux-gnu/gtk-3.0/3.0.0/immodules.cache`.
        Version Line:

            # Created by /usr/lib/i386-linux-gnu/libgtk-3-0t64/gtk-query-immodules-3.0 from gtk+-3.24.43

        **Failed to find fcitx5 in immodule cache at `/usr/lib/i386-linux-gnu/gtk-3.0/3.0.0/immodules.cache`**

        Found immodules cache for gtk `3.24.43` at `/usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules.cache`.
        Version Line:

            # Created by /usr/lib/x86_64-linux-gnu/libgtk-3-0t64/gtk-query-immodules-3.0 from gtk+-3.24.43

        **Failed to find fcitx5 in immodule cache at `/usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules.cache`**

        **Cannot find fcitx5 im module for gtk 3 in cache.**

4.  Gtk IM module files:

    1.  gtk 2:

        All found Gtk 2 immodule files exist.

    2.  gtk 3:

        All found Gtk 3 immodule files exist.

    3.  gtk 4:

        All found Gtk 4 immodule files exist.

# Configuration:
## Fcitx Addons:
1.  Addon Config Dir:

    Found fcitx5 addon config directory: `/usr/share/fcitx5/addon`.

2.  Addon List:

    1.  Found 21 enabled addons:

            Classic User Interface 5.1.11
            Clipboard 5.1.11
            DBus 5.1.11
            DBus Frontend 5.1.11
            Emoji 5.1.11
            Fcitx4 Frontend 5.1.11
            IBus Frontend 5.1.11
            Input method selector 5.1.11
            Keyboard 5.1.11
            KDE Input Method Panel 5.1.11
            Status Notifier 5.1.11
            Notification 5.1.11
            Quick Phrase 5.1.11
            Rime 5.1.8
            Spell 5.1.11
            Unicode 5.1.11
            DBus Virtual Keyboard 5.1.11
            Wayland 5.1.11
            Wayland Input method frontend 5.1.11
            XCB 5.1.11
            X Input Method Frontend 5.1.11

    2.  Found 0 disabled addons:

3.  Addon Libraries:

    All libraries for all addons are found.

4.  User Interface:

    Found 3 enabled user interface addons:

        Classic User Interface
        KDE Input Method Panel
        DBus Virtual Keyboard

## Input Methods:
1.  `/home/sangge/.config/fcitx5/profile`:

        [Groups/0]
        # Group Name
        Name=Default
        # Layout
        Default Layout=us
        # Default Input Method
        DefaultIM=rime

        [Groups/0/Items/0]
        # Name
        Name=keyboard-us
        # Layout
        Layout=

        [Groups/0/Items/1]
        # Name
        Name=rime
        # Layout
        Layout=

        [GroupOrder]
        0=Default

# Log:
1.  `date`:

        Wednesday, October 09, 2024 PM07:42:14 HKT

2.  `/home/sangge/.config/fcitx5/crash.log`:

    `/home/sangge/.config/fcitx5/crash.log` not found.

**Warning: the output of fcitx5-diagnose contains sensitive information, including the distribution name, kernel version, name of currently running programs, etc.**

**Though such information can be helpful to developers for diagnostic purpose, please double check and remove as necessary before posting it online publicly.**

**Comment by wengxt** [comment]
简繁切换是 Ctrl Shift F ，和一些程序中的搜索键相同。

我看你似乎没有安装 fcitx5-gtk fcitx5-qt ？这个应该是你不能用输入法的原因

alacrity 可以继续用 xim 所以没有问题。

乱码是你的字体不支持

**Reply by Smart-SangGe** [reply]
简繁切换我按ctrl+shift+f并没有切换？（alacritty内倒是弹出了搜索）。

下面这些是qt的库吗？我已经安装了的，就是前几天突然不能用了，之前都能用。
sudo apt install libfcitx5-qt6-1 libfcitx5-qt1 libfcitx5-qt-data
libfcitx5-qt6-1 is already the newest version (5.1.7-1).
libfcitx5-qt1 is already the newest version (5.1.7-1).
libfcitx5-qt-data is already the newest version (5.1.7-1).
libfcitx5-qt-data set to manually installed.
Summary:
  Upgrading: 0, Installing: 0, Removing: 0, Not Upgrading: 1

字体的话我是配置了nerd font，打字候选框都显示nerd font，emoji在终端也能正常显示，但是通过输入法打的就显示框框了。（其实我的问题是emoji不太出现在候选中，比如打“花”，需要我翻完生僻字才会有emoji的框框）

**Reply by Smart-SangGe** [reply, marked as answer]
在知乎中找到一个可行的方案，现在在其他软件中打字正常了。另外两个有空我在研究咋回事好了。
```
添加一个文件：~/.pam_environment
加入以下几行：
GTK_IM_MODULE DEFAULT=fcitx
QT_IM_MODULE DEFAULT=fcitx
XMODIFIERS DEFAULT=@im=fcitx
SDL_IM_MODULE DEFAULT=fcitx
```
https://zhuanlan.zhihu.com/p/508797663

**Reply by wengxt** [reply]
中文输入法下ctrl shift f才会切换
